crowbar_framework/app/models/service_object.rb

```diff
@@ -1405,6 +1420,17 @@ def apply_role(role, inst, in_queue)
     # XXX: This should not be done this way.  Something else should request this.
     system("sudo", "-i", Rails.root.join("..", "bin", "single_chef_client.sh").expand_path.to_s) if !ran_admin
 
+    # are there any roles to remove from the runlist?
+    role_to_remove = databag["deployment"][@bc_name]["elements"].select {|r| r =~ /_remove/}.keys.first
```

Comment:
I tested this with `heat` which has only one role. I will test it with `ceilometer` and update the PR accordingly